src/definition/populate_test.go

```diff
@@ -80,6 +80,76 @@ func metricsNamingManipulator(ms metric.MetricSet, entity sdk.Entity, clusterNam
 	return ms.SetMetric("displayName", entity.Name, metric.ATTRIBUTE)
 }
 
+func TestIntegrationProtocol2PopulateFunc_AddingNewMetricSetToEntity(t *testing.T) {
+	rawGroupsSample1 := RawGroups{
+		"test": {
+			"entity_id_1": RawMetrics{
+				"raw_metric_name_1": 1,
+			},
+		},
+	}
+	rawGroupsSample2 := RawGroups{
+		"test": {
+			"entity_id_1": RawMetrics{
+				"raw_metric_name_2": "foo",
+			},
+		},
+	}
+	specs := SpecGroups{
+		"test": SpecGroup{
+			TypeGenerator: fromGroupEntityTypeGuessFunc,
+			Specs: []Spec{
+				{"metric_1", FromRaw("raw_metric_name_1"), metric.GAUGE, true},
+				{"metric_2", FromRaw("raw_metric_name_2"), metric.ATTRIBUTE, true},
+			},
+		},
+	}
+	integration, err := sdk.NewIntegrationProtocol2("nr.test", "1.0.0", new(struct{}))
+	if err != nil {
+		t.Fatal()
+	}
```

Comment:
Perhaps `require.Nil()` can be used here? It seems odd to mix `assert` and regular error handling :thinking:

Reply:
definitely this was purely c&p :shame: